spec/features/user_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'rails_helper'
+
+RSpec.describe 'User', type: :feature do
+  let(:user) { User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') }
+  let(:post) { Post.new(author: user, title: 'Hello', text: 'This is my first post') }
+
+  before { user.save }
+  context 'index page' do
+    it 'show username of all other users' do
+      visit root_path
+      users = User.all
+      users.each do |user|
+        expect(page).to have_content(user.name)
+      end
+    end
+    it 'shows the profile picture for each user' do
+      visit root_path
+      users = User.all
+      users.each do |user|
+        expect(page).to have_selector("img[src='#{user.photo}']")
+      end
+    end
+    it 'shows the number of posts for each user' do
+      visit root_path
+      users = User.all
+      users.each do |user|
+        expect(page).to have_content("Number of posts: #{user.posts_counter}")
+      end
+    end
+
+    it "redirects me to the user's show page" do
+      visit root_path
+      users = User.all
+      users.each do |user|
+        click_link user.name
+        expect(page).to have_current_path(user_path(user))
+        visit root_path
+      end
+    end
+  end
+
+  context 'show page' do
+    it 'shows profile picture' do
+      user = User.first
+      visit user_path(user)
+      expect(page).to have_selector("img[src='#{user.photo}']")
+    end
+
+    it 'show user name' do
+      user = User.first
+      visit user_path(user)
+      expect(page).to have_content(user.name)
+    end
+
+    it 'show number of posts by the user' do
+      user = User.first
+      visit user_path(user)
+      expect(page).to have_content("Number of posts: #{user.posts_counter}")
+    end
+
+    it "shows a button that lets me view all of a user's posts." do
+      user = User.first
+      visit user_path(user)
+      expect(page).to have_selector('a', text: 'See all posts')
+    end
+  end
+
+  context 'show' do
```

Comment:
- [ ] **[OPTIONAL]** Both contexts are the same, they're describing the `show` page. I suggest you merge them, even include the one for the `clicks` as a nested context because it is also part of the show page and doing so would make more sense! 🤓 🔧 

- [ ] **[OPTIONAL]** Your contexts approach is fine, however, when you keep adding features to your pages, this test file is going to become really large. The common convention to separate these feature tests is in different files, for example, one for `user_index` and another for `user_show` pages. 🤓 🔧